config/src/main/java/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurer.java

```diff
@@ -0,0 +1,286 @@
+import org.springframework.security.web.access.intercept.RequestAuthorizationContext;
+import org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+import org.springframework.security.web.util.matcher.RequestMatcherEntry;
+import org.springframework.util.Assert;
+
+/**
+ * Adds a URL based authorization using {@link AuthorizationManager}.
+ *
+ * @param <H> the type of {@link HttpSecurityBuilder} that is being configured
+ * @author Evgeniy Cheban
+ */
+public final class AuthorizeHttpRequestsConfigurer<H extends HttpSecurityBuilder<H>>
+		extends AbstractHttpConfigurer<AuthorizeHttpRequestsConfigurer<H>, H> {
+
+	private final AuthorizationManagerRequestMatcherRegistry registry;
+
+	/**
+	 * Creates an instance.
+	 * @param context the {@link ApplicationContext} to use
+	 */
+	public AuthorizeHttpRequestsConfigurer(ApplicationContext context) {
+		this.registry = new AuthorizationManagerRequestMatcherRegistry(context);
+	}
+
+	/**
+	 * The {@link AuthorizationManagerRequestMatcherRegistry} is what users will interact
+	 * with after applying the {@link AuthorizeHttpRequestsConfigurer}.
+	 * @return the {@link AuthorizationManagerRequestMatcherRegistry} for further
+	 * customizations
+	 */
+	public AuthorizationManagerRequestMatcherRegistry getRegistry() {
+		return this.registry;
+	}
+
+	@Override
+	public void configure(H http) {
+		AuthorizationManager<HttpServletRequest> authorizationManager = this.registry.createAuthorizationManager();
+		AuthorizationFilter authorizationFilter = new AuthorizationFilter(authorizationManager);
+		http.addFilter(postProcess(authorizationFilter));
+	}
+
+	private AuthorizationManagerRequestMatcherRegistry addMapping(List<? extends RequestMatcher> matchers,
+			AuthorizationManager<RequestAuthorizationContext> manager) {
+		for (RequestMatcher matcher : matchers) {
+			this.registry.addMapping(matcher, manager);
+		}
+		return this.registry;
+	}
+
+	/**
+	 * Registry for mapping a {@link RequestMatcher} to an {@link AuthorizationManager}.
+	 *
+	 * @author Evgeniy Cheban
+	 */
+	public final class AuthorizationManagerRequestMatcherRegistry
+			extends AbstractRequestMatcherRegistry<AuthorizedUrl> {
+
+		private final DelegatingAuthorizationManager.Builder managerBuilder = DelegatingAuthorizationManager.builder();
+
+		private List<RequestMatcher> unmappedMatchers;
+
+		private AuthorizationManagerRequestMatcherRegistry(ApplicationContext context) {
+			setApplicationContext(context);
+		}
+
+		private void addMapping(RequestMatcher matcher, AuthorizationManager<RequestAuthorizationContext> manager) {
+			this.unmappedMatchers = null;
+			this.managerBuilder.add(new RequestMatcherEntry<>(matcher, manager));
+		}
+
+		private AuthorizationManager<HttpServletRequest> createAuthorizationManager() {
+			Assert.state(this.unmappedMatchers == null,
```

Comment:
While the `DelegatingAuthorizationManager` constructor will make sure there is at least one mapping, it might be helpful to the end user to see a more contextual error here like:

```
At least one mapping is required (for example, authorizeHttpRequests().anyRequest().authenticated())
```

This is similar to the error message thrown with `authorizeRequests`.

Reply:
I changed the message in the `DelegatingAuthorizationManager`'s constructor.

Reply:
This feels odd since if an application is using `DelegatingAuthorizationManager` independently from the DSL, then the error message will describe a remedy that doesn't apply.

Reply:
@jzheaux I added validation to `createAuthorizationManager `